src/Engine/ProtoCore/FFI/FFIExecutionManager.cs

```diff
@@ -36,6 +37,16 @@ public Assembly LoadAssembly(string name)
             {
                 try
                 {
+                    //if the assembly is named system.private.xyz -
+                    //then we can't load it, and it's very likely
+                    //already loaded, just return it.
+                    if (name.ToLower().Contains("system.private"))
+                    {
+                        //TODO cache to a map.
+                        return AppDomain.CurrentDomain.GetAssemblies()
+                        .Where(x => x.Location==name)
+                        .FirstOrDefault();
+                    }
```

Comment:
I thought this started to happen only in .net6. If so, shouldn't we have this code inside a .net6 framework conditional?

Reply:
we could, but I don't think this assembly even exists in net48. The conditional is probably safest though.